### PR TITLE
Feature add error detail interface - bug fix in 3.39.1

### PR DESCRIPTION
**Pull request by hnorrie16**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>io.electrum</groupId>
   <artifactId>service-interface-base</artifactId>
-  <version>3.39.1</version>
+  <version>3.39.2-RC2</version>
```
Comment (sowen123) [inline pom.xml:8]: Remember to remove

File: src/docs/devguide/hugo/site/content/release-notes.md [hunk 2]
```diff
@@ -1,6 +1,12 @@
 This page describes changes to the Service Interface Base API implemented across different releases of the interface.
 
-## 3.39.1
+## 3.39.2
+
+Released 01 June 2022
```
Comment (sowen123) [inline src/docs/devguide/hugo/site/content/release-notes.md:5]: Date reminder

File: src/docs/devguide/hugo/site/content/release-notes.md [hunk 3]
```diff
@@ -1,6 +1,12 @@
...
+
+* Bug fix: The argument type of the `detailMessage()` builder method in the `IErrorDetail` interface was incorrect. It has been changed to type `Object`
```
Comment (sowen123) [inline src/docs/devguide/hugo/site/content/release-notes.md:7]:
```suggestion
* Bug fix: The argument type of the `detailMessage()` builder method in the `IErrorDetail` interface was incorrect. It has been changed to a `Object` from a `String`
```

**Review by sowen123 — CHANGES_REQUESTED** [review]
Please add tests!

File: src/docs/devguide/hugo/site/content/release-notes.md (hunk 3 above)
Comment (hnorrie16) [reply src/docs/devguide/hugo/site/content/release-notes.md:7]: Fixed.

File: pom.xml (hunk 1 above)
Comment (hnorrie16) [reply pom.xml:8]: Removed.

File: src/docs/devguide/hugo/site/content/release-notes.md (hunk 2 above)
Comment (hnorrie16) [reply src/docs/devguide/hugo/site/content/release-notes.md:5]: Updated.

File: src/main/java/io/electrum/vas/example/model/ErrorDetail.java [hunk 4]
```diff
@@ -154,32 +157,36 @@ public void setErrorTypeFromString(String errorTypeString) throws Exception {
...
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("class ErrorDetail {\n");
+        sb.append("    id: ").append(Utils.toIndentedString(id)).append('\n');
+        sb.append("    errorType: ").append(Utils.toIndentedString(errorType)).append('\n');
+        sb.append("    errorMessage: ").append(Utils.toIndentedString(errorMessage)).append('\n');
+        sb.append("    detailMessage: ").append(Utils.toIndentedString(detailMessage)).append('\n');
+        sb.append("    originalId: ").append(Utils.toIndentedString(originalId)).append('\n');
+        sb.append('}');
+        return sb.toString();
```
Comment (sowen123) [inline src/main/java/io/electrum/vas/example/model/ErrorDetail.java:190]: Is your formatter wrong or was this wrong before? The toString has also changed here, why did you change it?

File: pom.xml [hunk 5]
```diff
@@ -5,7 +5,7 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>io.electrum</groupId>
   <artifactId>service-interface-base</artifactId>
-  <version>3.39.2-RC2</version>
+  <version>3.39.2-RC4</version>
```
[the remark this answers is not in the document]
Comment (hnorrie16) [reply pom.xml:8]: Removed.
I am unsure of how to go about this, am I meant to submit with the "RC" and then wait for releasing to remove it, or should I remove the "RC" from the get go?

File: src/docs/devguide/hugo/site/content/release-notes.md [hunk 6]
```diff
@@ -2,9 +2,9 @@ This page describes changes to the Service Interface Base API implemented across
 
 ## 3.39.2
 
-Released 01 June 2022
+Released 08 June 2022
```
[the remark this answers is not in the document]
Comment (hnorrie16) [reply src/docs/devguide/hugo/site/content/release-notes.md:5]: Fixed.

File: src/docs/devguide/hugo/site/content/release-notes.md [hunk 7]
```diff
@@ -2,9 +2,9 @@ This page describes changes to the Service Interface Base API implemented across
...
 
-* Bug fix: The argument type of the `detailMessage()` builder method in the `IErrorDetail` interface was incorrect. It has been changed to type `Object`
+* Bug fix: The argument type of the `detailMessage()` builder method in the `IErrorDetail` interface was incorrect. It has been changed to a `Object` from a `String`.
```
[the remark this answers is not in the document]
Comment (hnorrie16) [reply src/docs/devguide/hugo/site/content/release-notes.md:7]: Haha all good!
Fixed.

File: src/main/java/io/electrum/vas/example/model/ErrorDetail.java [hunk 8]
```diff
@@ -154,32 +157,36 @@ public void setErrorTypeFromString(String errorTypeString) throws Exception {
          throw new Exception("No matching error type for '" + errorTypeString + "' could be found.");
       }
    }
-
+    @ApiModelProperty(value = "")
```
[the remark this answers is not in the document]
Comment (hnorrie16) [reply src/main/java/io/electrum/vas/example/model/ErrorDetail.java:160]: Fixed.

**Review by hnorrie16 — COMMENTED** [review]
Review changed implemented.

**Review by sowen123 — APPROVED** [review]
LGTM :)

**Pull request merged**